File: src/main/java/me/austin/queer/util/chat/ChatHelper.java
package me.austin.queer.util.chat;

import me.austin.queer.TransRights;
import me.austin.queer.util.Globals;
import me.austin.queer.mixin.accessors.AccessorMinecraftClient;
import net.minecraft.text.Text;
import net.minecraft.util.Formatting;

public interface ChatHelper extends Globals {
    public static void addMessage(String message) {
        mc.inGameHud.getChatHud().addMessage(Text.of(message));
    }

    public static void addPreMessage(String message) {
        addMessage(Formatting.BOLD + "[" + TransRights.NAME + Formatting.AQUA + "]" + message + Formatting.WHITE);
    }

    public static void addErrorMessage(String message) {
        addMessage(Formatting.BOLD + "[ERROR]"  + Formatting.RED + message + Formatting.WHITE);
    }

    static void openChat() {
        ((AccessorMinecraftClient)mc).openChatScreen(mc.inGameHud.getChatHud().getMessageHistory().toString());
    }
}

File: src/main/kotlin/trans/rights/client/util/chat/ChatHelper.kt
package trans.rights.client.util.chat

import net.minecraft.text.Text
import trans.rights.client.TransRights.Companion.NAME
import trans.rights.client.misc.api.Globals

object ChatHelper : Globals {
    private const val prefix = "\\u00A79[${NAME}]\\u00A79"

    fun send(text: String) {
        minecraft.inGameHud.chatHud.addMessage(Text.of("$prefix$text"))
    }
}
